src/Symfony/Component/Security/Guard/GuardAuthenticatorInterface.php

```diff
@@ -39,11 +39,15 @@
      * Whatever value you return here will be passed to getUser() and checkCredentials()
      *
      * For example, for a form login, you might:
-     *
-     *      return array(
-     *          'username' => $request->request->get('_username'),
-     *          'password' => $request->request->get('_password'),
-     *      );
+     * 
+     *      if ($request->request->has('_username')) {
+     *          return array(
+     *              'username' => $request->request->get('_username'),
+     *              'password' => $request->request->get('_password'),
+     *          );
+     *      } else {
+     *          return;
+     *      }
```

Comment:
The else is not not necessary, the if statement around it is sufficient enough.

Question though, why would this create an infinite redirect?